src/Tribe/Importer/File_Importer_Events.php

```diff
@@ -443,22 +443,62 @@ private function get_timezone( $timezone_candidate ) {
 	}
 
 	/**
-	 * Returns the `post_excerpt` to use.
+	 * Get Description from Import or Existing Value.
 	 *
-	 * Will return the existing one if present.
+	 * @since TBD
 	 *
-	 * @param int $event_id
-	 * @param string $import_excerpt
+	 * @param int   $event_id The event id being updated by import.
+	 * @param array $record   An event record from the import.
 	 *
-	 * @return string
+	 * @return string The description value to update the event with.
 	 */
-	private function get_post_excerpt( $event_id, $import_excerpt ) {
-		if ( $event_id ) {
-			$post_excerpt = get_post( $event_id )->post_excerpt;
+	protected function get_description( $event_id, $record ) {
+
+		$import_exists = $this->has_value_by_key( $record, 'event_description' );
+
+		// if the description is not being imported and there is no id, return empty string.
+		if ( ! $import_exists && empty( $event_id ) ) {
+			return '';
+		}
+
+		// if the description is not being imported and there is an id, return current description.
+		if ( ! $import_exists && $event_id ) {
+			return get_post( $event_id )->post_content;
+		}
+
+		$import_description = $this->get_value_by_key( $record, 'event_description' );
+
+		// If there is no event id we return the imported description, even if empty.
+		return $import_description;
+	}
+
+	/**
+	 * Get Excerpt from Import or Existing Value.
+	 *
+	 * @since TBD
+	 *
+	 * @param int   $event_id The event id being updated by import.
+	 * @param array $record   An event record from the import.
+	 *
+	 * @return string The excerpt value to update the event with.
+	 */
+	protected function get_excerpt( $event_id, $record ) {
+
+		$import_exists = $this->has_value_by_key( $record, 'event_excerpt' );
+
+		// if the excerpt is not being imported and there is no id, return empty string.
+		if ( ! $import_exists && empty( $event_id ) ) {
+			return '';
+		}
 
-			return empty( $post_excerpt ) && ! empty( $import_excerpt ) ? $import_excerpt : $post_excerpt;
+		// if the excerpt is not being imported and there is an id, return current excerpt.
+		if ( ! $import_exists && $event_id ) {
+			return get_post( $event_id )->post_excerpt;
```

Comment:
No can do: `get_post` should return a `WP_Post` instance, but it might not and this will raise an exception or trigger an error.
Make sure, here and elsewhere, to check that calls to `get_post` or `tribe_get_event` are actually returning what you expect:

```php
$post = get_post( $post );
if( ! $post instanceof \WP_Post ) {
    // Not a post, bail.
}

// It's a post,  move on.

Reply:
I made the updates and merged the methods.